Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.h

```diff
@@ -24,17 +24,48 @@ namespace AZ::Dom
         //! For the path "/foo/bar" only "/foo/bar" will match while
         //! "/foo" and "/foo/bar/baz" will not.
         ExactPath,
-        //! The path, and any of its subpaths, will match.
-        //! For the path "/foo/bar" both "/foo/bar" and any subpaths like
-        //! "/foo/bar/0" will match, while "/foo" and orthogonal paths like
+        //! The path, and any of its parent paths, will match.
+        //! For the path "/foo/bar" both "/foo/bar" and any parent paths like
+        //! "/foo" will match, while "/foo/bar/0" and orthogonal paths like
         //! "/bar" will not
-        PathAndSubpaths,
-        //! Any of the path's subpaths will match, excepting the path itself.
-        //! For the path "/foo/bar", "/foo/bar" will not match but "/foo/bar/0"
+        PathAndParents,
+        //! Any of the path's parents will match, excepting the path itself.
+        //! For the path "/foo/bar", "/foo/bar" will not match but "/foo"
         //! will.
-        SubpathsOnly,
+        ParentsOnly,
     };
 
+    //! Defines traversal behavior when calling DomPrefixTree<T>::VisitPath.
+    enum class PrefixTreeTraversalFlags
+    {
+        //! No flags, will default to visiting all path-relative nodes in least to most specific order.
+        None = 0x0,
+        //! If set, any entries at the exact path specified to VisitPath will be ignored.
+        //! For the path "/foo" an entry at "/foo" will not match, while "/" or "/foo/0" may match, depending on the
+        //! other traversal flags.
+        ExcludeExactPath = 0x01,
+        //! If set, any parent entries to the path specified to VisitPath will be ignored.
+        //! For the path "/foo/0", "/" and "/foo" will not match, while "/foo/0" and "/foo/0/1" may match, depending
+        //! on the other traversal flags.
+        ExcludeParentPaths = 0x02,
+        //! If set, any child entries to the patch specified to VisitPath will be ignored.
+        //! For the path "/foo", "/foo/1" and "/foo/2" will not match, while "/foo" and "/" may match, depending
+        //! on the other traversal flags.
+        ExcludeChildPaths = 0x04,
+        //! If set, this visit operation will visit paths in a breadth-first manner.
```

Comment:
It looks like the documentation for these flags are inverted.

It isn't the default to `TraverseLeastToMostSpecific` using a bread first search?

Reply:
You're... absolutely right! The flags, funnily enough, are correctly implemented in the code, but the documentation / defaults are exactly wrong. Fixed (I think, my brain clearly can't be trusted to write docs for the prefix tree)